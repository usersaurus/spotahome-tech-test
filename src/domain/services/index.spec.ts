import { getHousesList } from './index'
jest.mock('../../api')

describe('The getHouseList service', () => {
  it('returns a House array', async () => {
    const housesListInfo = await getHousesList()

    expect(housesListInfo.houses).toBeInstanceOf(Array)
  })

  it('returns the correct info', async () => {
    const housesListInfo = await getHousesList()

    expect(housesListInfo.houses).toMatchSnapshot()
  })
})
