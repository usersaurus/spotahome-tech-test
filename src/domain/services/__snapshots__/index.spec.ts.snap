// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The getHouseList service returns the correct info 1`] = `
Array [
  Object {
    "address": "Paseo de la Castellana",
    "city": "madrid",
    "id": "79906",
    "image": "https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/74ef00db848c1ce44ea03677836b5bb61243e3f9d6a9d5a656d3c3cb.jpg",
    "link": "https://www.spotahome.com/madrid/for-rent:apartments/79906?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads",
    "title": "madrid - 280.000,00 €",
  },
  Object {
    "address": "calle virtudes",
    "city": "madrid",
    "id": "79994",
    "image": "https://d1052pu3rm1xk9.cloudfront.net/fsosw_960_540_verified_ur_6_50/9617b48286d8cf89fe755b19d884901be10f6b613a6fc872be2e83be.jpg",
    "link": "https://www.spotahome.com/madrid/for-rent:apartments/79994?utm_source=housinganywhere&utm_medium=cffeeds&utm_campaign=normalads",
    "title": "madrid - 95.000,00 €",
  },
]
`;
